ls1intum/Artemis
# `Programming exercises`: Allow users to open the repository in a separate browser window in the clone dialog

### Checklist
#### General


#### Client

### Motivation and Context
Following issue #4582 , a button was added to the clone repository menu, opening the repository in a separate browser window.

### Description
A button was added in the `clone.repro-button.component.html`, which will open the repository link in a separate browser window. If SSH is used, the button will not be displayed. The CSS-Styling has been added to a new `clone-repo-button.component.scss` file. 

- **Chrome and Chromium browsers**: No limitation, works with (click) as well as with (string+click)
- **Safari**: Clicking the button is disabled on the browser side to prevent the subconscious transfer of user information to external sides. This is limited by the browser and cannot be changed. With string+(click), the button works. 
- **Firefox**: The following message will be displayed when opening the new tap. This is issued by the browser as the authentication is done by the username contained in the repro-link. This cannot be changed. This message is **also** displayed if the user manually copies the repro-link and tries to open it in the same / new tap.
![156412162-1c8934d7-442c-434d-af67-3e6b6cb92803](https://user-images.githubusercontent.com/94070506/158151795-536abebf-9622-4d33-bb80-f01bab48f522.png)



### Steps for Testing

Prerequisites:
- 1 Student
- 1 Programming Exercise

1. Log in to Artemis
2. Navigate to Course with an existing programming exercise
3. Try to clone the repository. In the displayed menu there should be a button that opens the repository in a separate browser. If SSH is used, the button should not be displayed. 

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage






### Screenshots


![image](https://user-images.githubusercontent.com/94070506/156160806-ab79036e-4a1b-4d97-a485-daf22bdf728b.png)
![image](https://user-images.githubusercontent.com/94070506/156161473-ede6f59e-1cf8-47c5-99e3-fa6b78054d6f.png)
